sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -454,7 +454,20 @@ class JacksonParser(
       schema.getFieldIndex(parser.getCurrentName) match {
         case Some(index) =>
           try {
-            row.update(index, fieldConverters(index).apply(parser))
+            val value = try {
+              fieldConverters(index).apply(parser)
+            } catch {
+              case PartialResultException(partialResult, e) if enablePartialResults =>
```

Comment:
why do we swallow the exception earlier? what happens if we leave the exception to the outer `catch` block?

Reply:
Because it is more convenient to do it this way. Yes, we could leave it in the outer catch block but then I would have to copy 

```scala
row.update(index, value)
skipRow = structFilters.skipRow(row, index)
bitmask(index) = false
```
for each case of PartialResultException.

This is just a convenience thing to not duplicate the code.

Reply:
Can we have a common trait of `PartialResultException`, `PartialArrayDataResultException` and `PartialMapDataResultException`, so that we don't need to duplicate code?

Reply:
This is a good idea, let me make the change.

Reply:
I have updated the code but it is now more difficult to read in my opinion.